tools/models/architectures.py

```diff
@@ -0,0 +1,79 @@
+import torch.nn as nn
+from inSituML.loss_functions import EarthMoversLoss
+from inSituML.ks_models import INNModel
+
+
+class ModelFinal(nn.Module):
```

Comment:
Having a more generic concept of the model to be trained is great. Note though, this class remains a somewhat specific instance, i.e.  it has encoder, decoder, inner model. Please rename to something more descriptive, to respect this.